### PR TITLE
Qiskit backend for QCI

### PR DESCRIPTION
**Pull request by anpaz**

### CODE REVIEW

File: azure-quantum/azure/quantum/qiskit/backends/qci.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+from qiskit.providers import Options
+
+if TYPE_CHECKING:
+    from azure.quantum.qiskit import AzureQuantumProvider
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "QCISimulatorBackend"
```
Comment (ashwinmayya) [inline azure-quantum/azure/quantum/qiskit/backends/qci.py:23]: Adding a comment to ensure this doesn't get lost-
As per Andres, we need to add QCIQPUBackend here

Comment (ashwinmayya) [inline azure-quantum/azure/quantum/qiskit/backends/qci.py:23]: Adding a comment to ensure this doesn't get lost-
As per Andres, we need to add QCIQPUBackend here

File: azure-quantum/tests/unit/test_qiskit.py [hunk 2]
```diff
@@ -537,3 +537,82 @@ def test_qiskit_get_rigetti_qpu_targets(self):
...
+            assert qiskit_job._azure_job.details.input_params["arguments"] == []
+
+            # Make sure the job is completed before fetching the results
+            self._qiskit_wait_to_complete(qiskit_job, provider)
+
+            if JobStatus.DONE == qiskit_job.status():
+                result = qiskit_job.result()
+                print(result)
+                assert sum(result.data()["counts"].values()) == shots
+                assert np.isclose(result.data()["counts"]["[0, 0, 0]"], shots//2, 20)
```
Comment (ricardo-espinoza) [inline azure-quantum/tests/unit/test_qiskit.py:593]: Question out of curiosity, why are we not using regular division in `shots/2`?

File: azure-quantum/azure/quantum/qiskit/backends/qci.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        """Base class for interfacing with an QCI Simulator backend"""
+        default_config = BackendConfiguration.from_dict(
+            {
+                "backend_name": name,
+                "backend_version": __version__,
+                "simulator": True,
+                "local": False,
+                "coupling_map": None,
+                "description": "QCI simulator on Azure Quantum",
+                "basis_gates": QIR_BASIS_GATES,
```
Comment (ricardo-espinoza) [inline azure-quantum/azure/quantum/qiskit/backends/qci.py:64]: Cool, these are coming directly from the qiskit-qir package. :)

**Review by ricardo-espinoza — APPROVED** [review]

File: azure-quantum/tests/unit/test_qiskit.py (hunk 2 above)
Comment (anpaz) [reply azure-quantum/tests/unit/test_qiskit.py:593]: it's unclear to me too :)

**Pull request merged**